lib/elixir/test/elixir/kernel/warning_test.exs

```diff
@@ -28,6 +28,26 @@ defmodule Kernel.WarningTest do
     purge Sample
   end
 
+  test "unused variable in redefined function in different file" do
+    output = capture_err(fn ->
+      Code.eval_string """
+      defmodule Sample do
```

Comment:
So you can pass `:file` option to eval_string if that's the case. No need for a fixture file. :)